kura/org.eclipse.kura.container.provider/src/main/java/org/eclipse/kura/container/provider/ContainerInstance.java

```diff
@@ -95,6 +101,18 @@ public void deactivate() {
         logger.info("deactivate...done");
     }
 
+    public synchronized ContainerInstanceState getState() {
```

Comment:
What about returning just `this.state.getClass().getSimpleName()`? In this way adding or removing a state does not require keeping the enum `ContainerInstanceState` up to date. Something similar is done in log at line 140.

Reply:
I don't like passing around the `State` reference. What if a user attempts to change it/call a method that it shouldn't call?

Reply:
Yes, I was referring to just the name

Reply:
Addressed in 4bd4975eb71b45ac01da4ff0b1c42350b2b67339